addons/purchase_stock/tests/test_reordering_rule.py

```diff
@@ -1175,3 +1175,48 @@ def test_supplierinfo_last_purchase_date(self):
         replenishment_info = self.env['stock.replenishment.info'].create({'orderpoint_id': orderpoint.id})
         supplier_info = replenishment_info.supplierinfo_ids
         self.assertEqual(supplier_info.last_purchase_date, dt.today().date(), "The last_purhchase_date should be set to the most recent date_order from the purchase orders")
+
+    def test_propagate_updated_quantity_multi_step(self):
+        """ Ensure an auto-generated purchase order correctly adjusts associated picking
+        (esp. multi-step-reception) quantities of linked picking records when the relevant POL
+        `product_qty` is changed.
+        """
+        warehouse_1 = self.env['stock.warehouse'].search([('company_id', '=', self.env.user.id)], limit=1)
```

Comment:
The test is overly complicated for no real use:
- MTO is unnecessary for that case.
- It's very hacky to change the `procure_method` like that
- Avoid using `Form()` when you don't need the actual form (e.g. like checking an `onchange`). Meaning that if you want to create a product for your test, simply use `create()`.

Instead, you could simply do the following:
- Use the replenish wizard to trigger your procurement. Not only it will properly follow the client's use-case, it will be also cleaner as it's a supported operation :smile: 
- Edit the PO line & confirm the PO
- Check the related moves